src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Services/Implementation/ContextHandler.cs

```diff
@@ -18,14 +25,167 @@ namespace Altinn.Platform.Authorization.Services.Implementation
+                }
+
+                if (attribute.AttributeId.OriginalString.Equals(XacmlRequestAttribute.TaskAttribute))
+                {
+                    taskAttributeValue = attribute.AttributeValues.First().Value;
+                }               
+            }
+
+            bool resourceAttributeComplete = false;
+
+            if (!string.IsNullOrEmpty(orgAttributeValue) &&
+                !string.IsNullOrEmpty(appAttributeValue) &&
+                !string.IsNullOrEmpty(instanceAttributeValue) &&
+                !string.IsNullOrEmpty(resourcePartyAttributeValue) &&
+                !string.IsNullOrEmpty(taskAttributeValue))
+            {
+                // The resource attributes are complete
+                resourceAttributeComplete = true;
+            }
+            else if (!string.IsNullOrEmpty(orgAttributeValue) &&
+                !string.IsNullOrEmpty(appAttributeValue) &&
+                string.IsNullOrEmpty(instanceAttributeValue) &&
+                !string.IsNullOrEmpty(resourcePartyAttributeValue) &&
+                string.IsNullOrEmpty(taskAttributeValue))
+            {
+                // The resource attributes are complete
+                resourceAttributeComplete = true;
+            }
+
+            if (!resourceAttributeComplete)
+            {
+                if (!string.IsNullOrEmpty(instanceAttributeValue))
+                {
+                    Instance instanceData = await _policyInformationRepository.GetInstance(instanceAttributeValue);
+
+                    if (string.IsNullOrEmpty(orgAttributeValue))
+                    {
+                        resourceContextAttributes.Attributes.Add(GetAttribute(XacmlRequestAttribute.OrgAttribute, instanceData.Org));
+                    }
+
+                    if (string.IsNullOrEmpty(appAttributeValue))
+                    {
+                        string app = instanceData.AppId.Split("/")[1];
+                        resourceContextAttributes.Attributes.Add(GetAttribute(XacmlRequestAttribute.AppAttribute, app));
+                    }
+
+                    if (string.IsNullOrEmpty(taskAttributeValue))
+                    {
+                        resourceContextAttributes.Attributes.Add(GetAttribute(XacmlRequestAttribute.TaskAttribute, instanceData.Process.CurrentTask.ElementId));
+                    }
+
+                    if (string.IsNullOrEmpty(resourcePartyAttributeValue))
+                    {
+                        resourceContextAttributes.Attributes.Add(GetAttribute(XacmlRequestAttribute.PartyAttribute, instanceData.InstanceOwnerId));
+                    }
+
+                    resourcePartyAttributeValue = instanceData.InstanceOwnerId;
+                }
+            }
+
+            await EnrichSubjectAttributes(request, resourcePartyAttributeValue);
+        }
+
+        private XacmlAttribute GetAttribute(string attributeId, string attributeValue)
```

Comment:
Would createAttribute be a more suitable name here? 
Just judging by what the method does..

Reply:
If we are just creating yes, here it is more like forming the attribute and returning it. So i think this name is suitable